### ISSUE TITLE
Crash when recieving PCAPs

### ISSUE THREAD

**Issue by VaiTon** [eciavatta/caronte#28]
As I try to send PCAPs files (captured with `tcpdump -i ens18 -w out.pcap`) to caronte via API, either via POST to `/api/pcap/upload` or via POST to `/api/pcap/file` I get the following error and the docker container crashes:

```
caronte_1  | panic: interface conversion: interface is nil, not hyperscan.database
caronte_1  | 
caronte_1  | goroutine 117 [running]:
caronte_1  | github.com/flier/gohs/hyperscan.NewScratch(0x0, 0x0, 0x0, 0x0, 0x0)
caronte_1  |    /root/go/pkg/mod/github.com/flier/gohs@v1.0.0/hyperscan/runtime.go:22 +0x45
caronte_1  | main.(*BiDirectionalStreamFactory).takeScanner(0xc00031adc0, 0x0, 0x0, 0x0)
caronte_1  |    /caronte/connection_handler.go:117 +0x163
caronte_1  | main.(*BiDirectionalStreamFactory).New(0xc00031adc0, 0x1, 0x4, 0x4, 0x440f77a8, 0x0, 0xdb612e5d, 0x0, 0x4, 0x2, ...)
caronte_1  |    /caronte/connection_handler.go:175 +0xeec
caronte_1  | github.com/google/gopacket/tcpassembly.(*StreamPool).getConnection(0xc00012e380, 0x1, 0x4, 0x4, 0x440f77a8, 0x0, 0xdb612e5d, 0x0, 0x4, 0x2, ...)
caronte_1  |    /root/go/pkg/mod/github.com/google/gopacket@v1.1.17/tcpassembly/assembly.go:502 +0x1ea
caronte_1  | github.com/google/gopacket/tcpassembly.(*Assembler).AssembleWithTimestamp(0xc0000acb80, 0x1, 0x4, 0x4, 0x440f77a8, 0x0, 0xdb612e5d, 0x0, 0xc0001ae000, 0xdff7f00, ...)
caronte_1  |    /root/go/pkg/mod/github.com/google/gopacket@v1.1.17/tcpassembly/assembly.go:550 +0x27f
caronte_1  | main.(*PcapImporter).parsePcap(0xc000482700, 0xc000316b80, 0x40, 0xc0284a8bbad061e8, 0x363bcb01f2, 0x1580e20, 0x87b2, 0x0, 0x0, 0x0, ...)
caronte_1  |    /caronte/pcap_importer.go:249 +0xb8c
caronte_1  | created by main.(*PcapImporter).ImportPcap
caronte_1  |    /caronte/pcap_importer.go:132 +0x57e
```

**Comment by dorisroot1** [comment]
我尝试在docker中搭建并启动，可以访问页面，在页面中尝试上传pcap的文件失败，文件名字哪里是红色的，页面也没有数据，不知道你的是不是也这样。
![image](https://user-images.githubusercontent.com/30891737/121460599-c5e5e500-c9df-11eb-9366-63b33eb289fd.png)

**Comment by eciavatta** [comment]
This is not a problem related to the pcaps, but it is a bug of the library that compiles regular expressions.
If an invalid regex is added to the database of patterns, the application crashes when checking if some rules (e.g. the flag rule) are matched with the connection just extracted from the pcap.
This is a huge problem that needs to be fixed as soon as possibile, inserting a validation check before a regex is about to be added to the database.

@VaiTon which are the rules (the regexes) that you have been added before processing the pcap? I know that if the regex is wrapped around the `/`, Hyperscan, the regex library, crashes.

**Comment by VaiTon** [comment]
@eciavatta we tried changing the initial server address and it doesn't crash anymore.

In fact after replacing the public IP of the server with the LAN IP and sending some PCAPs captured with `-i lo` it worked.

Also cc @codesimo

**Comment by eciavatta** [comment]
Should be solved in 489ae5c7c65e7f60c3a0e5dc86b1e669e7bf67b5
